internal/autoevent/executor.go

```diff
@@ -86,7 +86,7 @@ func readResource(e *Executor, dic *di.Container) (event *dtos.Event, err errors
 	vars[common.Name] = e.deviceName
 	vars[common.Command] = e.sourceName
 
-	res, err := application.GetCommand(context.Background(), e.deviceName, e.sourceName, "", dic)
+	res, err := application.GetCommand(context.Background(), e.deviceName, e.sourceName, "", false, dic)
```

Comment:
@lenny-intel @cloudxxx8 

Go does not escape `.`(dot) seems like a big issue. For example I was testing with `.rotation` autoevent source name and core-data is unable to receive event correctly:
```shell
level=DEBUG ts=2023-03-21T09:00:48.169042468Z app=core-data source=subscriber.go:66 msg="Event received from MessageBus. Topic: edgex/events/device/device-simple/Simple-Device/Simple-Device01//rotation, Correlation-id: f877bd96-c375-4d96-8f4e-57187fc6b721"
level=ERROR ts=2023-03-21T09:00:48.169249433Z app=core-data source=subscriber.go:82 msg="event's profileName Simple-Device mismatches with the name Simple-Device01 received in topic"
```

I don't think let user URL-esacpe the autoevent source name a good idea.
Do you have an ideal solution for this ?

Reply:
no, auto event source name should not be encoded.

for dot, we need to use another encoding mechanism for message bus.  maybe base64.  it had better be supported by the golang library.

Reply:
@hahattan In the Mar-22-2023 Core WG meeting, we decided to create a wrapper of URL encoding maybe in go-mod-core-contracts to handle dot character specificly.
https://en.wikipedia.org/wiki/URL_encoding#Character_data
It could be encoded to `%2E`

Reply:
PR for the helper function: https://github.com/edgexfoundry/go-mod-core-contracts/pull/825